File: CVC_Tianming/codes/hyperspectral_image_processing/fusion/soft_isotropic.m
function xx = soft_isotropic(x,tau)

l = length(x)/2;

y = x(1:l);
z = x(l+1:end);

yz = sqrt(y.^2+z.^2+eps);
s = max(yz-tau,0);

y = (y./yz).*s;
z = (z./yz).*s;

xx = [y;z];

File: Hyperspectral_Image_Processing_python/fusion/soft_isotropic.py
import numpy as np
import sys

def soft_isotropic(x, tau):
    l = len(x) / 2

    y = x[:l]
    z = x[l+1:]

    yz = np.sqrt(y**2 + z**2 + sys.float_info.epsilon)
    s = max(yz - tau, 0)

    y = (y / yz) * s;
    z = (z / yz) * s;

    return [y,z]